docs/sdk/quickstart-by-language/quickstart-nodejs.mdx

```diff
@@ -0,0 +1,156 @@
+can do what, based on the policy that you defined.
+
+Each user, once his identity is verified by the authentication provider, gets issued a session. During this session, his unique id
+obtained from the JWT (JSON Web Token), which is synced to Permit, maps to a specific role within the Policy.
+
+Each role has a set of rules that were pre-defined, which a policy engine can understand, evaluate, and therefore return if the user
+is permitted or not, based on the user id, action and resource passed into the SDK function.
+
+You can run a permission check with permit.check(), passing in 3 arguments:
+
+- `user`: a unique string id that identifies the user doing the action (and the currently logged in user).
+- `action`: the action performed.
+- `resource`: the resource (object) the action is performed on.
+
+In the following example we are checking that a user with the unique id `john@smith.com` can `read` a `document`.
+
+```javascript
+const permitted = await permit.check("john@permit.io", "read", "document");
+
+if (permitted) {
+  console.log("John is PERMITTED to read a document");
+} else {
+  console.log("John is NOT PERMITTED to read a document");
+}
+```
+
+## Other basic functionality with the SDK
+
+In this quickstart, we'll introduce you to some core Permit.io functionality using the SDK. As you'll see, the SDK is designed to
+be simple and logical, making it easy to develop and manage policies.
+
+### Creating a resource
+
+A resource is the fundamental building block of a policy. It's what the user can interact with. It could be a 'document', a 'building'
+or even a 'patient health record'.
+
+```javascript
+await permit.api.resource.create({
+  key: "repository",
+  name: "Repository",
+  actions: {
+    clone: {},
+    push: {},
+    view: {},
+  },
+});
+```
+
+### Creating a new role with permissions
+
+A role defines a set of resources and the specific actions a user can perform on those resources. A user can only perform these
+actions if they have been assigned the corresponding role.
+
+````javascript
+await permit.api.roles.create({
+    key: "manager",
+    name: "Manager",
+    permissions: [
+      "repository:clone",
+      "repository:view",
+    ],
+  });```
+````
+
+### Assigning a role
+
+A role assignment, as the name suggests, assigns a specific role with defined permissions to a user identity. This role assignment
+is specific to a particular tenant.
+
+```javascript
+await permit.api.roleAssignments.assign({
+  user,
+  role,
+  tenant: "default",
+});
+```
+
+## Working with the basic SDK functionality
+
+For a comprehensive overview of basic SDK functionality, please refer to the essential list of SDK functions available
+[here](/category/nodejs).
+
+## Working with the full SDK specification
+
+As part of your IDE (Integrated Development Environment), we highly suggest that you install an extension (such as **Tabnine** or
```

Comment:
Code completion (or, as VSCode calls it, IntelliSense) is a feature that examines the AST of the project dependencies and proposes various options for each SDK the project uses. GitHub copilot is a product that uses AI for better completion where the traditional lacks. https://en.wikipedia.org/wiki/Code_completion

We shouldn't propose using Copilot as a recommendation for working with the SDK for two reasons:
1. It means our SDK lacks structure (which is not the case), and that it is not intuitive.
2. Copilot can propose SDK paths that do not exist or work at all.
See also the comment below regarding the whole section.